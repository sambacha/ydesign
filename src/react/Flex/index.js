import React from 'react';
import PropTypes from 'prop-types';
import Box from '../Box';

const Flex = React.forwardRef((props, ref) => <Box ref={ref} {...props} />);

Flex.defaultProps = {
  display: 'flex',
};

Flex.propTypes = {
  ...Box.propTypes,
};

Flex.displayName = 'Flex';

export default Flex;
