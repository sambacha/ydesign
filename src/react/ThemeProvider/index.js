import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
