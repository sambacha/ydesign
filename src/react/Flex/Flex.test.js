import Flex from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Flex component sanity', () => {
  it('has name', () => {
    expect(Flex.displayName).toBe('Flex');
  });

  it('matches default snapshot', () => {
    const component = render(<Flex />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Flex />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
