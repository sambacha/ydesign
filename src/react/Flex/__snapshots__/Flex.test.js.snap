// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex component sanity matches default snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Flex component sanity matches themed snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
  display="flex"
/>
`;
